Makefile

```
@@ -215,6 +216,9 @@ docs-preview: ## Preview the documentation through mkdocs
 docs-build:  ## Build the documentation to the `site/` directory
 	mkdocs build --strict
 
+generate-docs-crds: ## Build the documentation for CRD
```

This adds about 20s now every time I locally build with these changes. How about adding a condition that will only trigger this if the types.go changed?

---

Even with the changes, if someone is changing types.go, it will rebuild every time they are building locally. Even better if this would only build whenever the types.go actually changes. What about a condition such as this so the makefile will skip the action if the file hasn't changed? I may not have the right syntax, but something like this:
```suggestion
generate-docs-crds: pkg/apis/ceph.rook.io/v1/types.go ## Build the documentation for CRD
```

---

I added this script https://github.com/rook/rook/pull/12110/files#diff-2b6fe2b59a116565d7ee0824759b784384eeffb43b3281de18651053183e1ce9, which allow rebuild the docs only if types.go has changes otherwise it will skip.

The same script is executed on the CI to validate if there's no differences (for that reason I set the flag "--force").

Does it make sense?

---

The challenge even with that script is that even when types.go is edited, the developer will usually build many times to change something, then test it again, before the PR is merged. We really only need to generate the docs if the types.go file timestamp has changed since the last local build. This is what the makefile can check for us, and only trigger the generate-docs-crds action if the file timestamp was changed since the last build. 

For example, let's say I change types.go. The first time I build locally, then the docs should generate the new docs. Then I need to change other code to test it, so I build again, and again, and again. If I never change types.go again during my testing, I don't need to rebuild the docs every time. The additional 20s can add up when iterating during development, if it's not needed every time. So I'll see if I can find the syntax for the makefile to do this for us later today...

---

I think it is done with this change https://github.com/rook/rook/pull/12110/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R219 can you take a look?